tests/lib/cli-engine/cli-engine.js

```diff
@@ -3709,46 +3896,66 @@ describe("CLIEngine", () => {
         describe("don't ignore the entry directory.", () => {
             const root = getFixturePath("cli-engine/dont-ignore-entry-dir");
 
-            it("'executeOnFiles(\".\")' should not load config files from outside of \".\".", () => {
-                CLIEngine = defineCLIEngineWithInMemoryFileSystem({
-                    cwd: () => root,
+            /* eslint-disable-next-line func-style */
+            let cleanup = () => { };
+
+            afterEach(async () => {
+                await cleanup();
+                cleanup = () => { };
+            });
+
+            it("'executeOnFiles(\".\")' should not load config files from outside of \".\".", async () => {
+                const teardown = createCustomTeardown({
+                    cwd: root,
                     files: {
                         "../.eslintrc.json": "BROKEN FILE",
```

Comment:
I think that `cleanup` will not delete this file.

Reply:
Yeah, nice catch. Let me look at that.

Reply:
I verified that this file is still removed. The library does a `path.resolve()` for each file and uses the absolute path for operations, so it doesn't restrict to the root.